Update to mopidy-ext-template 2.0 and Mopidy 4.0

jodal opened the pull request.

jodal committed:
Add mopidy-ext-template v1.1.0 answers

jodal committed:
Update to mopidy-ext-template v2.0.0

jodal committed:
Format with ruff

girst commented:
Taking this as an opportunity to ask for an alpha release on pypi :)

Am 24. Februar 2025 09:26:44 MEZ schrieb Stein Magnus Jodal ***@***.***>:
>
>You can view, comment on, or merge this pull request online at:
>
>  https://github.com/mopidy/mopidy-mpd/pull/70
>
>-- Commit Summary --
>
>  * Add mopidy-ext-template v1.1.0 answers
>  * Update to mopidy-ext-template v2.0.0
>  * Format with ruff
>
>-- File Changes --
>
>    A .copier-answers.yml (8)
>    M .github/workflows/ci.yml (26)
>    M .github/workflows/release.yml (21)
>    M .gitignore (17)
>    A README.md (205)
>    D README.rst (178)
>    M pyproject.toml (102)
>    M src/mopidy_mpd/network.py (2)
>    M tests/protocol/__init__.py (18)
>    M tests/protocol/test_stored_playlists.py (4)
>    M tests/test_extension.py (4)
>
>-- Patch Links --
>
>https://github.com/mopidy/mopidy-mpd/pull/70.patch
>https://github.com/mopidy/mopidy-mpd/pull/70.diff
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mopidy/mopidy-mpd/pull/70
>You are receiving this because you are subscribed to this thread.
>
>Message ID: ***@***.***&gt;

jodal committed:
Remove test cases for Refs without URIs

These can no longer be constructed using the Ref.* helper methods, and
we'll probably soon require manually created `Ref` instances to always
have URIs too.

jodal committed:
Remove test case for blank dates

Blank dates are no longer allowed by the model layer.

jodal committed:
Use proper UUIDs for MusicBrainz IDs

Both in tests, and also support the model field being an UUID instance
instead of a str.

jodal committed:
Fix most new type errors

Most of the errors left will be solved if Mopidy core makes `uri` fields
on models required.

jodal committed:
Require Mopidy v4.0.0a4 to allow use of Ref.track() without name
